index.d.ts

```diff
@@ -142,7 +142,7 @@ declare namespace ShopifyBuy {
     export type CurrencyCode = "GBP" | "EUR" | "USD" | "CAD";
 
     export interface MoneyV2 {
-        amount: number;
+        amount: number | string;
```

Comment:
Question -- I was dealing with this too. I found that it seems to just coalesce it to a number for me, does that not work in this case?

Reply:
The type change here alone doesn't fix the issue, as this is really just for the transpiler to check before build time. However, at runtime, the old business logic in the consuming packages had assumed the field to be a `number` but instead, iwas receiving a `string`, which led to `NaN`s in a bunch of places. 

Ultimately, it's the responsibility of the consuming module to verify field types at runtime, especially if they're received via a network request. By increasing the type accuracy here, it forces the developer to account for both `number` and `string` in case the actual data at runtime is invalid. This is why we sometimes have to perform type checks at runtime despite also having type definitions written in TypeScript (see [juniper-notifications example](https://github.com/hellojuniper-com/juniper-notifications/blob/558b377c985b66c40fbbd3a4c7ba713276dd4f0b/src/handlers/PostNotifications.ts#L8-L50)).